s2core/src/main/scala/org/apache/s2graph/core/Edge.scala

```diff
@@ -381,6 +645,23 @@ object Edge {
       propsWithoutLastDeletedAt.forall { case (_, v) => v.ts <= lastDeletedAt }
     }
 
+  def allPropsDeleted(props: Props): Boolean =
+    if (!props.containsKey(LabelMeta.lastDeletedAt.name)) false
+    else {
+      val lastDeletedAt = props.get(LabelMeta.lastDeletedAt.name).ts
+      props.remove(LabelMeta.lastDeletedAt.name)
+//      val propsWithoutLastDeletedAt = props
+//
+//      propsWithoutLastDeletedAt.forall { case (_, v) => v.ts <= lastDeletedAt }
+      var ret = true
+      val iter = props.entrySet().iterator()
+      while (iter.hasNext) {
+        val e = iter.next()
+        if (e.getValue.ts > lastDeletedAt) ret = false
```

Comment:
The while statement does not terminate the loop even if the condition '(e.getValue.ts> lastDeletedAt)' is satisfied.